### PR TITLE
WGS pipeline re-org effort 1

### PR DESCRIPTION
**Pull request by SHuang-Broad**
@kvg and @ewanbroad 

This is an updated WGS pipeline that brings several utilities and DeepVariantPepper.

Given that this PR is kind of big, I broke it into several commits, with each commit as orthogonal to others as much as possible, so that the review process is easier.

@kvg, I still need to test run this on a sample to check for obvious errors (talk about tests...), so I'm making it as a draft.
But I don't think there will be structural changes.

Thanks,
Steve

### CODE REVIEW

**Comment by SHuang-Broad** [comment]
@kvg, this has been tested with CCS data, but the ONT test data keeps getting all kinds of issues form the tools themselves. It should be ready by tomorrow.

**Comment by SHuang-Broad** [comment]
@kvg @ewanbroad  this is now ready for review. The [CCS, ONT] workflows have been tested on real data (ONT with high coverage sample).

One known issue is regarding the DV-Margin-Pepper pipeline: for CCS data, it's not producing phased VCF yet, but I do have a branch where it's finally working, so I'll bring in the changes in a later, separate PR.

Thanks.

**Review by kvg — APPROVED** [review]
Looks fantastic! I put a few comments in, but there's nothing major in them that would prevent the PR from being merged immediately.

My only major comment is that we should think about how we handle reference sequences and interval list information. But we can do that in a subsequent PR.

File: wdl/PBCLRWholeGenome.wdl [hunk 1]
```diff
@@ -49,7 +64,18 @@ workflow PBCLRWholeGenome {
...
+        input:
+            aligned_bam = bam,
+            aligned_bai = bai,
+            ref_fasta   = ref_map['fasta'],
+            bed_to_compute_coverage = bed_to_compute_coverage
+    }
+
     if (call_variants) {
+        # verify arguments are provided
+        if (! defined(fast_less_sensitive_sv)) {call Utils.StopWorkflow as fast_less_sensitive_sv_not_provided {input: reason = "Unprovided arg fast_less_sensitive_sv"}}
```
[the remark this answers is not in the document]
Comment (SHuang-Broad) [reply wdl/PBCLRWholeGenome.wdl:77]: That's certainly an option.
The original intention was to implement a simplistic argument verification mechanism, e.g.
* arg B is required when arg/flag A is specified
* arg A and arg C are mut. ex.
* etc.

File: wdl/tasks/CallVariantsONT.wdl [hunk 2]
```diff
@@ -1,41 +1,176 @@
...
+
+        Boolean call_small_vars_on_mitochondria
+        File? sites_vcf
+        File? sites_vcf_tbi
+
+        Boolean run_dv_pepper_analysis
+        Int? dvp_threads
+        Int? dvp_memmory
+        File? ref_scatter_interval_list_locator
+        File? ref_scatter_interval_list_ids
```
[the remark this answers is not in the document]
Comment (SHuang-Broad) [reply wdl/tasks/CallVariantsONT.wdl:36]: Totally agree.
As we discussed, this needs careful design, as it's hard to balance between competing hard/soft constraints.

File: wdl/tasks/CallVariantsONT.wdl [hunk 3]
```diff
@@ -1,41 +1,176 @@
...
+            }
+            
+            call ONTPepper.Pepper {
+                input:
+                    bam           = size_balanced_scatter.subset_bam,
+                    bai           = size_balanced_scatter.subset_bai,
+                    ref_fasta     = ref_fasta,
+                    ref_fasta_fai = ref_fasta_fai,
+                    threads       = select_first([dvp_threads]),
+                    memory        = select_first([dvp_memmory])
```
[the remark this answers is not in the document]
Comment (SHuang-Broad) [reply wdl/tasks/CallVariantsONT.wdl:133]: done

File: wdl/tasks/CallVariantsPBCCS.wdl [hunk 4]
```diff
@@ -1,40 +1,173 @@
...
+                input:
+                    bam           = size_balanced_scatter.subset_bam,
+                    bai           = size_balanced_scatter.subset_bai,
+                    ref_fasta     = ref_fasta,
+                    ref_fasta_fai = ref_fasta_fai,
+
+                    pepper_threads = select_first([dvp_threads]),
+                    pepper_memory  = select_first([dvp_memmory]),
+                    dv_threads = select_first([dvp_threads]),
+                    dv_memory  = select_first([dvp_memmory])
```
[the remark this answers is not in the document]
Comment (SHuang-Broad) [reply wdl/tasks/CallVariantsPBCCS.wdl:137]: done

**Comment by SHuang-Broad** [comment]
Regarding your comments about handling references better, particularly relevant to this PR:

* scatter-gather
* special contig names

I agree completely, and to do this, we need to 
* better understand where the references are used in the genome pipelines and 
* how references are used there

**Pull request merged**